### PR TITLE
[DRAFT] Issue #1254 Optimization for ReadFileResponse

### PR DESCRIPTION
**Pull request by scirin**
Added [Optimization for ReadFileResponse ](https://github.com/metalbear-co/mirrord/issues/1254)
Awaiting for your feedback for review!

+Apologies for such a delay

### CODE REVIEW

File: mirrord/agent/src/file.rs [hunk 1]
```diff
@@ -362,34 +367,37 @@ impl FileManager {
...
+        .get_mut(&fd)
+        .ok_or(ResponseError::NotFound(fd))
+        .and_then(|remote_file| {
+            if let RemoteFile::File(file) = remote_file {
+                let mut buffer = vec![0; buffer_size as usize];
+
+                let read_amount = file.read_at(&mut buffer, start_from)?;
+
+                // Further optimization: Slice the buffer based on the actual number of bytes read
+                let read_bytes = buffer[..read_amount].to_vec();
```
Comment (DmitryDodzin) [inline mirrord/agent/src/file.rs:387]: `.to_vec` is copying buffer slice into another vector, similar desired result can be achieved with `.truncate(read_amount)` where the `buffer` Vec should shrink to expected size (same for read function)

**Review by DmitryDodzin — CHANGES_REQUESTED** [review]
Thank you very much!

Looks great I've left a small nitpick comment and would ask to add a "changelog" file, we use [towncrier](https://github.com/twisted/towncrier) so it's simply a markdown file under `./changelog.d/1254.fixed.md` with a comment for the changelog (Similar to what is now in [./changelog.d](https://github.com/metalbear-co/mirrord/tree/b72720cbd3d113fc247e4a74b45eaa5dda07563c/changelog.d))
🚀

File: mirrord/agent/src/file.rs (hunk 1 above)
Comment (scirin) [reply mirrord/agent/src/file.rs:387]: Will fix, thank you for feedback!

**Comment by aviramha** [comment]
Hey, any update on this? :)

**Comment by scirin** [comment]
> Hey, any update on this? :)

will push today! sorry for delay

**Comment by aviramha** [comment]
Continued in #1601 (had no permissions to push to OP repo)

**Pull request closed without merging**